---
layout: archive
title: Zulip Chat Archive
permalink: /stream/113488-general/topic/mathlib.20CI.20submodule.20error.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/113488-general/index.html">general</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html">mathlib CI submodule error</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="284901163"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284901163" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284901163">(Jun 03 2022 at 16:23)</a>:</h4>
<p>Does anyone understand where the following error comes from?<br>
<a href="https://github.com/leanprover-community/mathlib/runs/6729106015?check_suite_focus=true">https://github.com/leanprover-community/mathlib/runs/6729106015?check_suite_focus=true</a></p>



<a name="284901179"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284901179" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284901179">(Jun 03 2022 at 16:23)</a>:</h4>
<p>ping <span class="user-mention" data-user-id="488744">@Jack M</span></p>



<a name="284901280"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284901280" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284901280">(Jun 03 2022 at 16:24)</a>:</h4>
<p>I suspect it's some issue with <span class="user-mention" data-user-id="488744">@Jack M</span> 's git setup, but we can't figure out what the issue is.</p>



<a name="284905891"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284905891" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284905891">(Jun 03 2022 at 17:03)</a>:</h4>
<p>The PR  <a href="https://github.com/leanprover-community/mathlib/pull/14542">#14542</a> has the same error. It seems that both <span class="user-mention" data-user-id="488744">@Jack M</span> and <span class="user-mention" data-user-id="488747">@Michael Blyth</span> have some issue with their setup. Does anyone have any ideas?</p>



<a name="284907981"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284907981" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Sebastian Ullrich <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284907981">(Jun 03 2022 at 17:21)</a>:</h4>
<p>It looks like some impurity from the runner machine to me, i.e. not your fault</p>



<a name="284908128"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284908128" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284908128">(Jun 03 2022 at 17:22)</a>:</h4>
<p>I mentioned this in the maintainer stream.</p>



<a name="284912063"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284912063" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Reid Barton <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284912063">(Jun 03 2022 at 17:56)</a>:</h4>
<p>On lines 309-310 the log says</p>
<div class="codehilite" data-code-language="Text only"><pre><span></span><code>  /usr/bin/git reset --hard HEAD
  HEAD is now at d60657c3e Added the concept of a line in projective space.
</code></pre></div>
<p>which is the commit with the submodule</p>



<a name="284913121"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284913121" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Reid Barton <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284913121">(Jun 03 2022 at 18:05)</a>:</h4>
<p>I suspect that whatever runner tried to build that commit with a submodule is now stuck</p>



<a name="284914729"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284914729" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284914729">(Jun 03 2022 at 18:18)</a>:</h4>
<p>Aha, so the current CI run on master made it past the checkout step, which is promising.<br>
<a href="https://github.com/leanprover-community/mathlib/runs/6730598604?check_suite_focus=true">https://github.com/leanprover-community/mathlib/runs/6730598604?check_suite_focus=true</a></p>
<p>It does indeed look like the issue is with a single runner.</p>



<a name="284918240"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284918240" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Gabriel Ebner <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284918240">(Jun 03 2022 at 18:44)</a>:</h4>
<p>Did somebody do something to fix the error? It looks like nael is fine again.</p>



<a name="284918360"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284918360" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Yaël Dillies <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284918360">(Jun 03 2022 at 18:45)</a>:</h4>
<p>Oh yes, hello, I am fine!</p>



<a name="284918629"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284918629" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Gabriel Ebner <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284918629">(Jun 03 2022 at 18:47)</a>:</h4>
<p>I was talking about <em>n</em>ael the VM, not Yaël the person. <span aria-label="smile" class="emoji emoji-1f642" role="img" title="smile">:smile:</span></p>



<a name="284918791"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284918791" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284918791">(Jun 03 2022 at 18:49)</a>:</h4>
<p>The only thing I did was delete the (empty) folder from same branch as the commit that Reid mentioned and pushed to that same branch</p>



<a name="284918873"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mathlib%20CI%20submodule%20error/near/284918873" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Adam Topaz <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/mathlib.20CI.20submodule.20error.html#284918873">(Jun 03 2022 at 18:50)</a>:</h4>
<p><a href="https://github.com/leanprover-community/mathlib/tree/projective_linear_subspace">branch#projective_linear_subspace</a></p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>